rest_api/common/src/paging.rs

```diff
@@ -0,0 +1,168 @@
+
+    Paging {
+        current: current_link,
+        offset,
+        limit,
+        total: query_count,
+        first: first_link,
+        prev: previous_link,
+        next: next_link,
+        last: last_link,
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    const TEST_LINK: &str = "/api/test?";
+
+    #[test]
+    fn test_default_paging_response() {
+        // Create paging response from default limit, default offset, a total of 1000
+        let test_paging_response = get_response_paging_info(None, None, TEST_LINK, 1000);
+        let generated_paging_response =
+            create_test_paging_response(DEFAULT_OFFSET, DEFAULT_LIMIT, 100, 0, 900);
+        assert_eq!(test_paging_response, generated_paging_response);
+    }
+
+    #[test]
+    fn test_50offset_paging_response() {
+        // Create paging response from default limit, offset of 50, and a total of 1000
+        let test_paging_response = get_response_paging_info(None, Some(50), TEST_LINK, 1000);
+        let generated_paging_response = create_test_paging_response(50, DEFAULT_LIMIT, 150, 0, 900);
+        assert_eq!(test_paging_response, generated_paging_response);
+    }
+
+    #[test]
+    fn test_550offset_paging_response() {
+        // Create paging response from default limit, offset value of 150, and a total of 1000
+        let test_paging_response = get_response_paging_info(None, Some(550), TEST_LINK, 1000);
+        let generated_paging_response =
+            create_test_paging_response(550, DEFAULT_LIMIT, 650, 450, 900);
+        assert_eq!(test_paging_response, generated_paging_response);
+    }
+
+    #[test]
+    fn test_950offset_paging_response() {
+        // Create paging response from default limit, offset value of 950, and a total of 1000
+        let test_paging_response = get_response_paging_info(None, Some(950), TEST_LINK, 1000);
+        let generated_paging_response =
+            create_test_paging_response(950, DEFAULT_LIMIT, 900, 850, 900);
+        assert_eq!(test_paging_response, generated_paging_response);
+    }
+
+    #[test]
+    fn test_50limit_paging_response() {
+        // Create paging response from default limit, offset of 50, and a total of 1000
+        let test_paging_response = get_response_paging_info(Some(50), None, TEST_LINK, 1000);
+        let generated_paging_response = create_test_paging_response(DEFAULT_OFFSET, 50, 50, 0, 950);
+        assert_eq!(test_paging_response, generated_paging_response);
+    }
+
+    #[test]
+    fn test_50limit_150offset_paging_response() {
+        // Create paging response from limit of 50, offset of 150, and total of 1000
+        let test_paging_response = get_response_paging_info(Some(50), Some(150), TEST_LINK, 1000);
+        let generated_paging_response = create_test_paging_response(150, 50, 200, 100, 950);
+        assert_eq!(test_paging_response, generated_paging_response);
+    }
+
+    fn create_test_paging_response(
```

Comment:
Feels like this could be PagingBuilder code and Paging fields non-pub so that the integrity of the struct is always valid.

Reply:
Fixed